src/Polly.Core/ResilienceContext.cs

```diff
@@ -55,6 +58,14 @@ private ResilienceContext()
     /// </summary>
     public ResilienceProperties Properties { get; } = new();
 
+    /// <summary>
+    /// Gets the collection of resilience events that occurred while executing the resilience strategy.
+    /// </summary>
+    /// <remarks>
+    /// If the number of resilience events is greater than zero it's an indication that the execution was unhealthy.
+    /// </remarks>
```

Comment:
Might people write their own extensions that report successful events? For example I was thinking earlier that maybe for when I look at caching I'd see at adding cache hit/miss telemetry.

Reply:
Inside your strategy you can use `ResilienceStrategyTelemetry` to report individual events and these will be reported here. However, these should really be treated as "warnings". 

In custom strategies you can add your own collection into `ResilienceProperties` and use that to report cache events.

For example:
``` csharp
var events = resilienceContext.Properties.GetValue(new ResiliencePropertyKey<List<string>>("cache-events"), new List<string>());
events.Add("hit");
```

Reply:
Also, maybe we can expand the `ReportedResilienceEvent` with `Severity` later?

That way you can also report `informational` events and use the common-infra.

Reply:
That sounds like a good idea.

Reply:
Created a new issue: #1143